MapboxNavigation/CarPlayNavigationViewController.swift

```diff
@@ -167,15 +169,18 @@ public class CarPlayNavigationViewController: UIViewController {
     public override func viewSafeAreaInsetsDidChange() {
         super.viewSafeAreaInsetsDidChange()
         
-        if let previousSafeAreaInsets = previousSafeAreaInsets {
-            let navigationBarIsOpen = view.safeAreaInsets > previousSafeAreaInsets
-            mapView?.compassView.isHidden = navigationBarIsOpen
+        defer {
+            previousSafeAreaInsets = view.safeAreaInsets
+            view.setNeedsUpdateConstraints()
         }
         
-        previousSafeAreaInsets = view.safeAreaInsets
+        // TODO: Verify `?? 0` should result in navigationBarIsOpen = false initially
+        let navigationBarIsOpen = view.safeAreaInsets.top > previousSafeAreaInsets?.top ?? 0
```

Comment:
Is there a possibility that this method could be called twice with the same top safe area inset, perhaps due to a race condition? If so, this check would be unreliable.

In practice, the top safe area inset is zero when the navigation bar is hidden and nonzero when the navigation bar is shown, correct? What if we check the absolute value rather than the value relative to a previous value?

Reply:
> Is there a possibility that this method could be called twice

👍 that was the case. `insets.top > 0` is a reliable way with the current setup to get the state of the nav bar.